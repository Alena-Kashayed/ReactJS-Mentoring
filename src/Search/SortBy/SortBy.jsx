import classNames from 'classnames/bind';
import PropTypes from 'prop-types';
import React from 'react';
import styles from './SortBy.scss';

function SortBy(props) {
  const cx = classNames.bind(styles);
  return props.movieNumber ? (
    <div className={styles.sortByGroup}>
      <span className={styles.sortBy}>Sort by</span>
      <button className={cx('releaseDay', 'sortByBtn')}>release day</button>
      <button className={cx('rating', 'sortByBtn', 'active')}>rating</button>
    </div>
  ) : null;
}

SortBy.propTypes = {
  movieNumber: PropTypes.number.isRequired,
};

export default SortBy;
