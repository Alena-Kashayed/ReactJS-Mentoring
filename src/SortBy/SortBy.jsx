import classNames from 'classnames/bind';
import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

import styles from './SortBy.scss';
import { setSortByReleaseDay, setSortByRating } from '../actions';

const cx = classNames.bind(styles);
const handleSortByReleaseDay = (dispatch) => {
  dispatch(setSortByReleaseDay);
};
const handleSortByRating = (dispatch) => {
  dispatch(setSortByRating);
};

const SortBy = ({ currentFilm, sortBy, dispatch }) => (
  !(Object.keys(currentFilm).length) ?
    (
      <div className={styles.sortByGroup}>
        <span className={styles.sortBy}>Sort by</span>
        <button
          onClick={() => handleSortByReleaseDay(dispatch)}
          className={cx('releaseDay', 'sortByBtn', { active: sortBy === 'releaseDay' })}
        >
          release day
        </button>
        <button
          onClick={() => handleSortByRating(dispatch)}
          className={cx('rating', 'sortByBtn', { active: sortBy === 'rating' })}
        >
          rating
        </button>
      </div>
    ) :
    null
);

SortBy.propTypes = {
  currentFilm: PropTypes.shape({
    category: PropTypes.string,
    director: PropTypes.string,
    mediatype: PropTypes.number,
    poster: PropTypes.string,
    rating: PropTypes.string,
    release_year: PropTypes.string,
    runtime: PropTypes.number,
    show_cast: PropTypes.string,
    show_id: PropTypes.number,
    show_title: PropTypes.string,
    summary: PropTypes.string,
    unit: PropTypes.number,
  }),
  dispatch: PropTypes.func.isRequired,
  sortBy: PropTypes.string,
};

SortBy.defaultProps = {
  currentFilm: null,
  sortBy: '',
};

const mapStateToProps = state => ({
  sortBy: state.sortBy.value,
  currentFilm: state.currentFilm,
});

export default connect(mapStateToProps)(SortBy);
