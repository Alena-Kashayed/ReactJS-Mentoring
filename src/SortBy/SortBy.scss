.sortByGroup {
  min-width: 240px;
  display: flex;
  justify-content: space-between;

  .sortBy {
    font-size: 16px;
    color: #525252;
  }

  .sortByBtn {
    border: none;
    background-color: transparent;
    font-weight: 600;
    font-size: 16px;
    color: #525252;

    &.active {
      color: #F55263;
    }
  }

  .releaseDay {

  }

  .rating {

  }
}
