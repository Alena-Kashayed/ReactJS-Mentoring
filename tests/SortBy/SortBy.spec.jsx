import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import SortBy from '../../src/SortBy/SortBy';

const mockStore = configureStore();

describe('SortBy', () => {
  it('is rendered correctly', () => {
    const wrapper = shallow(<SortBy
      store={mockStore({
        currentFilm: {},
        sortBy: { value: '' },
      })}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
