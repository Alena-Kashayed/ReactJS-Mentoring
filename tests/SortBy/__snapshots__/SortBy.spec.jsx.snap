// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBy is rendered correctly 1`] = `
<SortBy
  currentFilm={Object {}}
  dispatch={[Function]}
  sortBy=""
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
